File: python/algorithms/sorting/insertion_sort/insertion_sort.py
# algorithms/sorting/insertion_sort/insertion_sort.py
def insertion_sort(arr):
    
    for i in range(1,len(arr)):
        
        key = arr[i] # current element
        j = i - 1

        # move the elements biggest than key from position in the front
        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1
        
        arr[j + 1] = key

    return arr

File: csharp/algorithms/sorting/insertion_sort/InsertionSort.cs
namespace Algorithms.Sorting;

/// <summary>
/// Insertion Sort (in-place).
/// Time: O(n^2) worst/avg, O(n) best. Space: O(1)
/// Proof: see algorithms/sorting/insertion_sort/README.md
/// </summary>
public static class InsertionSort
{
    public static void Sort(int[] arr)
    {
        for (int i = 1; i < arr.Length; i++)
        {
            int key = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > key)
            {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }
    }
}